components/nimbus/src/stateless/cirrus_client.rs

```diff
@@ -133,27 +135,39 @@ impl CirrusClient {
         let nimbus_id = Uuid::new_v4();
         let available_randomization_units =
             AvailableRandomizationUnits::with_client_id(client_id.as_str());
-        let th = NimbusTargetingHelper::new(targeting_attributes);
+        let ta = TargetingAttributes::new(self.app_context.clone(), request_context);
+        let th = NimbusTargetingHelper::new(ta);
         let enrollments_evolver =
             EnrollmentsEvolver::new(&nimbus_id, &available_randomization_units, &th);
+        let state = self.state.lock().unwrap();
```

Comment:
Does this panic at runtime if it fails?

Reply:
Yep, it's worth noting a few things about that though.

1. It's our general paradigm used throughout the Nimbus client, as well as generally what is expected when using Mutexes according to the docs.
    * nimbus use: https://github.com/mozilla/application-services/blob/main/components/nimbus/src/nimbus_client.rs#L93
2. As long as I'm understanding the docs correctly, `lock` is going to block the current thread until it's able to acquire the Mutex. So if it's not available immediately it should be okay but I don't think we'll run into that very often in Python. 
3. FWIW I keep seeing the Python tests run in .5s, and it's running enrollment 1000 times plus some other stuff so I personally am not concerned. Happy to write some threaded tests if that'd be preferred

Mutex docs: https://doc.rust-lang.org/std/sync/struct.Mutex.html#method.lock

The docs _seem_ to imply that panics only happen if you try to lock it a second time on the same thread, but I'm not entirely certain that's what the "panic" block in the docs means. I'll do some research and get back to you on that